### ISSUE TITLE
No price information was found for ProductVariant ID "8" in the Channel "british-isles"

### ISSUE THREAD

**Issue by seminarian** [vendurehq/vendure#2440]
**Describe the bug**
`No price information was found for ProductVariant ID "8" in the Channel "british-isles".` Error message when loading products in subchannel `british-isles`
![image](https://github.com/vendure-ecommerce/vendure/assets/9881519/48384cf7-2049-4775-9726-15dcc10f01e3)

The cause was Product assigned to `british-isles` which has a custom field of type relation which has PVs configured that are not assigned to `british-isles` channel.

In our case I think the channel-aware custom fields would actually be the solution to use. Because up sell variants are actually channel dependent.

Product variant is not assigned to the channel:
![image](https://github.com/vendure-ecommerce/vendure/assets/9881519/2e037d9a-fdb1-4d09-93c6-eb5266a16b0f)

But that specific PV was assigned to a Product assigned to `british-isles`
```
product -> upsell variants custom field
2 -> 18 20
9 -> 2 18 20
14 -> 8 14
15 -> 2 11 20
16 -> 2 11 18

aggregated list 2 8 11 14 18 20

actually assigned PVs to channel:
2 11 17 18 19 20 21


**lacking pvs assigned to channel**: 8 14
```

**To Reproduce**
1. Add the following custom field config
```
    config.customFields.Product.push({
      name: 'upsellVariants',
      label: [{ languageCode: LanguageCode.en, value: 'Upsell variants' }],
      type: 'relation',
      entity: ProductVariant,
      list: true,
    });
```
2. Generate the migrations and restart dev server
3. In default channel add a few upsell variants to a product.
4. In a subchannel add the Product to the channel but make sure some of the upsell variants are not assigned to the subchannel.
5. Go to products page in admin UI for the subchannel

**Environment (please complete the following information):**
 - @vendure/core version: 2.1.0-next.4
 - Nodejs version: 16.9.0
 - Database (mysql/postgres etc): Postgres

**Comment by yasserlens** [comment]
We're also seeing the same error

**Comment by seminarian** [comment]
> We're also seeing the same error after upgrading from v2.0.4 to v2.1.4. Nothing else has changed that could have caused the error to appear. The error we have is:
> ```
> No price information was found for ProductVariant ID "99" in the Channel "default_channel”
> ```

The DefaultVariantPriceStrategy inside the core is not falling back anymore if a currency was requested which there is not a price for in the channel.
Check that.

Or override the strategy with your own version which suits your business needs.

**Comment by yasserlens** [comment]
Hello!
I'm not sure I follow re:"currency was requested which there is not a price for in the channel" - could you please provide an example?
Note that this is the default channel that it complains about - which, as far as I know, has access to all channel-specific variants/prices/currencies..etc.

**Comment by yasserlens** [comment]
tldr; this happened in a development instance, so to save time I solved this by deleting the offending Products.

I did a quick investigation before that and I found:
- The following query shows that the product variants the query complains about do exist for the channel when running the following query (channelId = 1 which is the default channel where I had the issue)
```sql
select * from "product_variant_price" where "channelId" = 1;
```
- However, the product variant price had the wrong currency associated with it in that table, it was "EUR" while the default channel has a "USD" defaultCurrencyCode
- I manually changed the currencyCode to USD. This solved the error messages for the default channel (#1) but raised new errors for another channel which had the product variants it was complaining about

In hindsight, I could have removed the association of those product variants to the channels that have currency issues then re-assigned them to those channels, hoping that would resolve the currency issues, but I didn't think of this at the time. Deleting the products seemed easier since that's all done in a development instance.
